test/README.md

```diff
@@ -84,10 +84,70 @@ ssh and web.  See the "Helpful tips" section below.
 
 ## Pixel tests
 
-The verify test suite contains ["pixel tests"](https://cockpit-project.org/blog/pixel-testing.html).
-Make sure to create the test/reference submodule before running tests which contain pixel tests.
+Pixel tests in Cockpit ensure that updates of our dependencies or code changes
+don't break unexpectedly break the UI, for example slight changes of layout,
+padding, color and everything which isn't easily tested by a human. They also
+give us confidence that an update of our UI Framework doesn't introduce changes
+in how Cockpit looks.
+
+Pixel tests are in essence quite simple, they make a screenshot of the current
+state and compare this to a known good state, if there is a difference the test
+fails and a pixel difference is shown. This works as our tests run in the
+[cockpit/tasks container](https://quay.io/repository/cockpit/tasks) which pins
+the browser and font rendering so repeated runs provide the same pixels. To
+generate new pixels this tasks container should be used as your own browser and
+font rendering software might generate different results. For more information
+read the ["introduction blog post"](https://cockpit-project.org/blog/pixel-testing.html).
+
+The test images are stored in git submodule in the `test/reference` directory and be fetched with:
+
+```sh
+./test/common/pixel-tests fetch
+```
+
+As Cockpit tests under multiple distributions and it is not worth the effort to
+run pixel tests on every supported distribution we only run them for the
+configured image configured in `test/reference-image`.
+
+Our tests don't know when it is interesting to make a screenshot of the current
+state, so our tests have to specifically assert for saved pixels using
+`browser.assert_pixels`. This assertion method requires an CSS selector to be
+provided and a title of the image. Pixel tests are generated in five layouts by
+default; desktop, medium, mobile, dark and rtl.
+
+Take a screenshot the content in `#detail-content`:
+```python
+browser.assert_pixels("#detail-content", "filesystem")
+```
+
+Take a screenshot the content in `#detail-content` and ignore all elements with
+a class `disk-stats` as they change per test run.
+```python
+browser.assert_pixels("#detail-content", "filesystem", ignore=[".disks-stats"])
+```
+
+Take a screenshot the content in `#detail-content` and skip it for a specific
+layout as it generates unstable pixels.
+```python
+browser.assert_pixels("#detail-content", "filesystem", skip_layouts=["rtl"])
+```
+
+To update pixel tests it's recommended to either start use the tasks container
+to generate specific new pixels or create a draft PR and let the tests run for
+`test/reference-image` and afterwards fetching the new pixels:
+
+```
+./test/common/pixel-tests fetch "https://cockpit-logs.us-east-1.linodeobjects.com/<snip>/log.html
```

Comment:
Last time I checked it didn't work with /log.html -- that's the bit that you have to cut off. Maybe that was fixed since then?

Reply:
Yes, that's fixed.